crates/sui/src/keytool.rs

```diff
@@ -804,6 +821,58 @@ impl KeyToolCommand {
                 .await?;
                 CommandOutput::ZkLoginSignAndExecuteTx(ZkLoginSignAndExecuteTx { tx_digest })
             }
+
+            KeyToolCommand::ZkLoginSigVerify {
+                sig,
+                tx_bytes,
+                provider,
+                curr_epoch,
+            } => {
+                match GenericSignature::from_bytes(
+                    &Base64::decode(&sig).map_err(|e| anyhow!("Invalid base64 sig: {:?}", e))?,
+                )? {
+                    GenericSignature::ZkLoginAuthenticator(zk) => {
+                        if tx_bytes.is_none() || provider.is_none() {
+                            return Ok(CommandOutput::ZkLoginSigVerify(format!(
+                                "Parsed zkLogin signature: {:?}",
+                                zk
+                            )));
+                        }
+                        println!("Parsed zkLogin signature: {:?}", zk);
+
+                        let tx_data: TransactionData = bcs::from_bytes(
+                            &Base64::decode(&tx_bytes.unwrap())
+                                .map_err(|e| anyhow!("Invalid base64 tx data: {:?}", e))?,
+                        )?;
+                        let client = reqwest::Client::new();
+                        let parsed: ImHashMap<JwkId, JWK> = fetch_jwks(
+                            &OIDCProvider::from_str(&provider.clone().unwrap()).unwrap(),
```

Comment:
Not sure how this JWK works, but consider handling the unwraps properly to avoid any panics.

Reply:
revised, the top level unwrap is handled at line 845

Reply:
Thanks for taking care of those unwraps!